### DISCUSSION TITLE
Components required for building from source

### DISCUSSION THREAD

**Discussion by Richardthegreatdev** [Macad3D/Macad3D#66] [Q&A]
when `restore` command is run, the following error occured:

Required installation of VisualStudio 2022 not found.
Please ensure that the following components are installed:
  Microsoft.VisualStudio.Workload.ManagedDesktop
  Microsoft.VisualStudio.Workload.NativeDesktop
  Microsoft.NetCore.Component.SDK
  Microsoft.VisualStudio.Component.Roslyn.Compiler
  Microsoft.VisualStudio.Component.Roslyn.LanguageServices
  Microsoft.Component.MSBuild
  Microsoft.VisualStudio.Component.TextTemplating
  Microsoft.VisualStudio.Component.ManagedDesktop.Core
  Microsoft.VisualStudio.Component.VC.CoreIde
  Microsoft.VisualStudio.Component.VC.Tools.x86.x64
  Microsoft.VisualStudio.Component.ManagedDesktop.Prerequisites
  Microsoft.VisualStudio.Component.VC.Redist.14.Latest
  Microsoft.VisualStudio.Component.VC.CLI.Support

For compiling Macad3D, which components are needed in order to avoid installing everything? This errors come out of installation with only `dot net desktop development` and `desktop development with c++` workloads installed in vs 2022 version 17.9.2. 

Thanks
Richard

**Comment by dipts** [comment]
The components listed are all those that are checked. It seems that Visual Studio itself or the VSWHERE utility program was not found. 

Please try to open and build the solution Macad3D.sln directly with Visual Studio.
